lib/spack/llnl/util/lock.py

```diff
@@ -780,6 +782,12 @@ class LockLimitError(LockError):
 class LockTimeoutError(LockError):
     """Raised when an attempt to acquire a lock times out."""
 
+    def __init__(self, lock_type, path, time, attempts):
+        msg = "Timed out waiting for a {} lock after {} and {} attempts on file: {}".format(
+            lock_type, pretty_seconds(time), attempts, path
+        )
+        super(LockTimeoutError, self).__init__(msg)
```

Comment:
Can you pass the attempts and filename as a second argument?  That'll make it look like this:

```console
==> Error: Timed out waiting for a write lock after 1.200ms.
    Made 4 attempts on file: /path/to/file
```

which means the path is less likely to wrap around.